src/_cffi_src/build_openssl.py

```diff
@@ -46,7 +49,16 @@ def _extra_compile_args(platform):
     When we drop support for CRYPTOGRAPHY_OPENSSL_LESS_THAN_110 we can
     revisit this.
     """
-    if platform not in ["win32", "hp-ux11", "sunos5"]:
+    # make sure the compiler used supports the flags to be added
+    is_gcc = False
+    if get_default_compiler() == "unix":
+        d = dist.Distribution()
+        cmd = config(d)
+        cmd._check_compiler()
+        is_gcc = ("gcc" in cmd.compiler.compiler[0] or
+                  "clang" in cmd.compiler.compiler[0])
+    if is_gcc or not (platform in ["win32", "hp-ux11", "sunos5"] or
+                      platform.startswith("aix")):
```

Comment:
Why does this use startswith? Different AIX releases have different suffixes?

Reply:
Hi @reaperhulk . Sorry I did not reply much earlier. Answer is yes, until recently the value included the version (e..g., 5. 6 or 7) of the AIX system it was built on (not the version it is running on). iirc, starting with Python 3.8 it only returns "aix". So, for now, still have to look at the "startswith" part.